include/engine/api/route_api.hpp

```diff
@@ -228,42 +228,50 @@ class RouteAPI : public BaseAPI
 
         std::vector<util::json::Object> annotations;
 
-        if (parameters.annotations_type != RouteParameters::AnnotationsType::None)
+        if (parameters.annotations_type != RouteParameters::AnnotationsType::None ||
+            parameters.annotations == true)
```

Comment:
I think there are _technically_ cases where only checking for `parameters.annotations` could be wrong. Say with a constructor where `annotations` defaults to `false` and then annotations_type is manually set.